enterprise/backend/src/metabase_enterprise/dependencies/api.clj

```diff
@@ -168,26 +168,119 @@
+   [:data [:map]]
+   [:dependents_count [:maybe [:ref ::usages]]]
+   [:errors {:optional true} [:set [:ref ::lib.schema.validate/error]]]])
+
+(defn- fields-for [entity-key]
+  ;; these specs should really use something like
+  #_[:data [:select-keys [:ref :blah/table] (entity-keys :table)]]
+  ;; but :select-keys seems to mess up open-api spec generation
+  (into [:map]
+        (map (fn [key] [key {:optional true} :any]))
+        (entity-keys entity-key)))
+
+(mr/def ::table-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/table]
+    [:type [:= :table]]
+    [:data (fields-for :table)]]])
+
+(mr/def ::card-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/card]
+    [:type [:= :card]]
+    [:data (fields-for :card)]]])
+
+(mr/def ::snippet-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/snippet]
+    [:type [:= :snippet]]
+    [:data (fields-for :snippet)]]])
+
+(mr/def ::transform-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/transform]
+    [:type [:= :transform]]
+    [:data (fields-for :transform)]]])
+
+(mr/def ::dashboard-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/dashboard]
+    [:type [:= :dashboard]]
+    [:data (fields-for :dashboard)]]])
+
+(mr/def ::document-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/document]
+    [:type [:= :document]]
+    [:data (fields-for :document)]]])
+
+(mr/def ::sandbox-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/sandbox]
+    [:type [:= :sandbox]]
+    [:data (fields-for :sandbox)]]])
+
+(mr/def ::segment-entity
+  [:merge ::base-entity
+   [:map
+    [:id ::lib.schema.id/segment]
+    [:type [:= :segment]]
+    [:data (fields-for :card)]]])
+
+(mr/def ::entity
+  [:multi {:dispatch :type}
+   [:table [:ref ::table-entity]]
+   [:card [:ref ::card-entity]]
+   [:snippet [:ref ::snippet-entity]]
+   [:transform [:ref ::transform-entity]]
+   [:dashboard [:ref ::dashboard-entity]]
+   [:document [:ref ::document-entity]]
+   [:sandbox [:ref ::sandbox-entity]]
+   [:segment [:ref ::segment-entity]]])
+
+(mu/defn- entity-value :- ::entity
+  [entity-type {:keys [id] :as entity} usages errors]
+  (cond-> {:id id
+           :type entity-type
+           :data (->> (select-keys entity (entity-keys entity-type))
+                      (m/map-vals format-subentity))
+           :dependents_count (usages [entity-type id])}
+    errors (assoc :errors (get errors [entity-type id]))))
 
 (def ^:private entity-model
-  {:table     :model/Table
-   :card      :model/Card
-   :snippet   :model/NativeQuerySnippet
+  {:table :model/Table
+   :card :model/Card
+   :snippet :model/NativeQuerySnippet
    :transform :model/Transform
    :dashboard :model/Dashboard
-   :document  :model/Document
-   :sandbox   :model/Sandbox
-   :segment   :model/Segment})
+   :document :model/Document
+   :sandbox :model/Sandbox
+   :segment :model/Segment})
+
+(def ^:private card-types
```

Comment:
I feel like this is the sort of thing that 100% is defined elsewhere and even so 100% should not be defined here anyway

Check `queries.schema` and/or `lib.schema.metadata`